llvm/test/CodeGen/AIE/GlobalISel/addrspace-lowering.mir

```diff
@@ -0,0 +1,161 @@
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2023-2024 Advanced Micro Devices, Inc. or its affiliates
+# RUN: llc -mtriple aie2 -run-pass=aie-addrspace-flattening %s -verify-machineinstrs -o - | FileCheck %s
+
+
+# Simple case where a pointer is defined by G_CONSTANT
+---
+name:            p5_CONSTANT
+legalized:       false
+tracksRegLiveness: true
+body:             |
+  bb.1:
+    ; CHECK-LABEL: name: p5_CONSTANT
+    ; CHECK: [[C:%[0-9]+]]:_(p0) = G_CONSTANT i20 0
+    ; CHECK-NEXT: [[LOAD:%[0-9]+]]:_(<32 x s8>) = G_LOAD [[C]](p0) :: (load (<32 x s8>) from `ptr addrspace(5) null`, addrspace 5)
+    %1:_(p5) = G_CONSTANT i20 0
+    %0:_(<32 x s8>) = G_LOAD %1:_(p5) :: (load (<32 x s8>) from `ptr addrspace(5) null`, addrspace 5)
+...
+
+
+# Simple case where a pointer is defined by G_IMPLICIT_DEF
+---
+name:            p5_IMPLICIT_DEF
+legalized:       false
+tracksRegLiveness: true
+body:             |
+  bb.1:
+    ; CHECK-LABEL: name: p5_IMPLICIT_DEF
+    ; CHECK: [[DEF:%[0-9]+]]:_(p0) = G_IMPLICIT_DEF
+    ; CHECK-NEXT: [[LOAD:%[0-9]+]]:_(<32 x s8>) = G_LOAD [[DEF]](p0) :: (load (<32 x s8>) from `ptr addrspace(5) null`, addrspace 5)
+    %1:_(p5) = G_IMPLICIT_DEF
+    %0:_(<32 x s8>) = G_LOAD %1:_(p5) :: (load (<32 x s8>) from `ptr addrspace(5) null`, addrspace 5)
+...
+
+# Simple case where a pointer is defined by G_ADDRSPACE_CAST
+---
+name:            p5_G_ADDRSPACE_CAST
+legalized:       false
+tracksRegLiveness: true
+body:             |
+  bb.1:
+    ; CHECK-LABEL: name: p5_G_ADDRSPACE_CAST
+    ; CHECK: [[C:%[0-9]+]]:_(p0) = G_CONSTANT i20 0
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(p0) = COPY [[C]](p0)
+    ; CHECK-NEXT: [[LOAD:%[0-9]+]]:_(<32 x s8>) = G_LOAD [[COPY]](p0) :: (load (<32 x s8>) from `ptr addrspace(5) null`, addrspace 5)
+    %10:_(p0) = G_CONSTANT i20 0
+    %1:_(p5) = G_ADDRSPACE_CAST %10:_(p0)
+    %0:_(<32 x s8>) = G_LOAD %1:_(p5) :: (load (<32 x s8>) from `ptr addrspace(5) null`, addrspace 5)
+...
+
+# Case where a pointer is defined by chain of G_ADDRSPACE_CAST
+---
+name:            chain_G_ADDRSPACE_CAST
+legalized:       false
+tracksRegLiveness: true
+body:             |
+  bb.1:
+    ; CHECK-LABEL: name: chain_G_ADDRSPACE_CAST
+    ; CHECK: [[C:%[0-9]+]]:_(p0) = G_CONSTANT i20 0
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(p0) = COPY [[C]](p0)
+    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:_(p0) = COPY [[COPY]](p0)
+    ; CHECK-NEXT: [[LOAD:%[0-9]+]]:_(<32 x s8>) = G_LOAD [[COPY1]](p0) :: (load (<32 x s8>) from `ptr addrspace(5) null`, addrspace 5)
+    %10:_(p0) = G_CONSTANT i20 0
+    %1:_(p5) = G_ADDRSPACE_CAST %10:_(p0)
+    %2:_(p6) = G_ADDRSPACE_CAST %1:_(p5)
+    %0:_(<32 x s8>) = G_LOAD %2:_(p6) :: (load (<32 x s8>) from `ptr addrspace(5) null`, addrspace 5)
```

Comment:
Nit: I think `ptr addrspace(6) null`, addrspace 6` would be better in the MMO, as we are loading using `p6`